data_diff/__init__.py

```diff
@@ -1,9 +1,86 @@
-from typing import Tuple
+from typing import Tuple, Iterator
 
 from .database import connect_to_uri
-from .diff_tables import TableSegment, TableDiffer
+from .diff_tables import (
+    TableSegment,
+    TableDiffer,
+    DEFAULT_BISECTION_THRESHOLD,
+    DEFAULT_BISECTION_FACTOR,
+    DbKey,
+    DbTime,
+    parse_table_name,
+)
 
 
-def create_source(db_uri: str, table_name: str, key_column: str, extra_columns: Tuple[str, ...] = ()):
-    db = connect_to_uri(db_uri)
-    return TableSegment(db, (table_name,), key_column, tuple(extra_columns))
+class TableRef:
+    """Connects to a URI and creates a table ref
+
+    This is an auxiliary class for diff_tables()
+    """
+
+    def __init__(self, db_uri: str, table_name: str):
+        self.db = connect_to_uri(db_uri)
+        self.table_name = table_name
+
+    def create_table_segment(self, **kwargs):
+        return TableSegment(self.db, parse_table_name(self.table_name), **kwargs)
+
+
+def diff_tables(
+    table1: TableRef,
+    table2: TableRef,
```

Comment:
Do you want to make this an iterator/list right away?

Reply:
No, I just think it's probably going to be some day.

Either way, using the args like `(db, table, db, table)` isn't good practice imho, and leads to confusion.